src/main/java/org/truffleruby/core/exception/CoreExceptions.java

```diff
@@ -573,6 +578,11 @@ public DynamicObject nameErrorUnknownIdentifier(TruffleObject receiver, Object n
 
     @TruffleBoundary
     public DynamicObject nameError(String message, Object receiver, String name, Node currentNode) {
+        return nameError(message, receiver, context.getSymbolTable().getSymbol(name), currentNode);
+    }
+
+    @TruffleBoundary
+    public DynamicObject nameError(String message, Object receiver, Object name, Node currentNode) {
```

Comment:
Could you be type `name` as DynamicObject and assert it's a Symbol?

Reply:
In this spec, the name is a ruby String:
https://github.com/ruby/ruby/blob/5a8a2706d33849cf42cd56714f519108a4ededfc/spec/ruby/core/exception/name_spec.rb#L46